### PR TITLE
test: Add dependency graph marshal tests

### PR DESCRIPTION
**Pull request by KirtiRamchandani**
Problem
Dependency graph and SBOM resource structs are newer API payloads, but their JSON marshal behavior was not covered by the resource-level marshal tests tracked by #55.

Root cause
The existing dependency graph tests focus on service request/response behavior. They do not round-trip the shared SBOM, dependency snapshot, manifest, detector, job, and creation response structs directly.

Solution
Add focused marshal tests for empty and populated dependency graph resource types, including SBOM metadata, package external refs, relationships, snapshot manifests, resolved dependencies, and snapshot creation data.

Tests run
- gofmt -w github/dependency_graph_marshal_test.go
- go test ./github -run 'Test(SBOM|CreationInfo|RepoDependencies|PackageExternalRef|SBOMRelationship|SBOMInfo|DependencyGraphSnapshot.*)_Marshal'
- go test ./github
- go test ./...
- git diff --cached --check

Linked issue
Part of #55

### CODE REVIEW

File: github/dependency_graph_marshal_test.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
```
Comment (gmlewis) [inline github/dependency_graph_marshal_test.go]: Instead of creating a new file for this, please delete this file and add the tests to the end of the already-existing `dependency_graph_test.go`.

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by KirtiRamchandani** [comment]
Thanks, moved the marshal tests into the existing dependency_graph_test.go file and removed the standalone test file. Reran the focused marshal test group, go test ./github, go test ./..., and git diff --check.

File: github/dependency_graph_test.go [hunk 2]
```diff
@@ -76,3 +76,363 @@ func TestDependencyGraphService_GetSBOM(t *testing.T) {
 		return resp, err
 	})
 }
+
+func TestSBOM_Marshal(t *testing.T) {
+	testJSONMarshal(t, &SBOM{}, `{}`)
```
Comment (Not-Dhananjay-Mishra) [inline github/dependency_graph_test.go:81]: Please add `t.Parallel()` to this and the other tests otherwise, linter will complain.
```suggestion
func TestSBOM_Marshal(t *testing.T) {
	t.Parallel()
	testJSONMarshal(t, &SBOM{}, `{}`)
```

File: github/dependency_graph_test.go [hunk 3]
```diff
@@ -76,3 +76,363 @@ func TestDependencyGraphService_GetSBOM(t *testing.T) {
...
+	testJSONMarshal(t, &CreationInfo{}, `{}`)
+
+	created := Timestamp{time.Date(2024, time.January, 2, 3, 4, 5, 0, time.UTC)}
+	u := &CreationInfo{
+		Created:  &created,
+		Creators: []string{"GitHub", "go-github"},
+	}
+
+	want := `{
+		"created": "2024-01-02T03:04:05Z",
```
Comment (Not-Dhananjay-Mishra) [inline github/dependency_graph_test.go:118]: Can we use `referenceTime` and `referenceTimeStr` for consistency?
Same applies for other tests.
```suggestion
	u := &CreationInfo{
		Created:  &Timestamp{referenceTime},
		Creators: []string{"GitHub", "go-github"},
	}

	want := `{
		"created": ` + referenceTimeStr + `,
```

**Comment by gmlewis** [comment]
Thank you for your understanding, @KirtiRamchandani.
Closing as no longer needed. See #55.
